enterprise/frontend/src/metabase-enterprise/data-studio/common/hooks/use-load-table-with-metadata/use-load-table-with-metadata.ts

```diff
@@ -1,20 +1,41 @@
+import { useEffect } from "react";
+
 import { skipToken, useGetTableQueryMetadataQuery } from "metabase/api";
+import Tables from "metabase/entities/tables";
+import { useDispatch } from "metabase/lib/redux";
 import type { TableId } from "metabase-types/api";
 
-export function useLoadTableWithMetadata(tableId: TableId | undefined) {
+type UseLoadTableWithMetadataOptions = {
+  includeForeignTables?: boolean;
+};
+
+export function useLoadTableWithMetadata(
+  tableId: TableId | undefined,
+  { includeForeignTables = false }: UseLoadTableWithMetadataOptions = {},
+) {
+  const dispatch = useDispatch();
+
   const {
     data: table,
-    isLoading,
+    isLoading: isTableLoading,
     error,
   } = useGetTableQueryMetadataQuery(
     tableId != null
       ? { id: tableId, include_editable_data_model: true }
       : skipToken,
   );
 
+  useEffect(() => {
+    if (!includeForeignTables || tableId == null) {
+      return;
+    }
+
+    dispatch(Tables.actions.fetchMetadataAndForeignTables({ id: tableId }));
```

Comment:
- we could use the RTK query directly
- we need to pass `include_editable_data_model` here as well

we could use `initiate` with the same logic as in https://github.com/metabase/metabase/blob/b933496d37eb69ab958f709dddd05b5fa4648cfd/frontend/src/metabase/entities/tables.js#L414 to load all fk tables
`initiate` example https://github.com/metabase/metabase/blob/4dbd4f25f68fd188553f0b695febe5510bd50edd/frontend/src/metabase/common/components/Pickers/MiniPicker/utils.ts#L66